src/lib/services/access-service.ts

```diff
@@ -699,7 +747,18 @@ export class AccessService {
             );
         }
 
-        return this.roleStore.delete(id);
+        const existingRole = await this.roleStore.get(id);
+        const existingPermissions = await this.store.getPermissionsForRole(id);
+        await this.roleStore.delete(id);
+        this.eventService.storeEvent({
+            type: ROLE_DELETED,
+            createdBy: deletedBy,
+            preData: {
+                ...existingRole,
+                permissions: this.sanitizePermissions(existingPermissions),
```

Comment:
how do we decide when to add extra data. This event duplicates the information from ROLE_CREATED. If we wanted to have "normalized" event we'd only provide minimum necessary information. In this case this is the role being deleted. Is it for convenience that we show the same data in ROLE_CREATED and ROLE_DELETED?

Reply:
Not sure, because data may have changed, we can trace back what the permissions were by the time of deleting this, but for that matter, we would only need the id when we delete any resource, and we're not doing that currently.